platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -1184,14 +1189,14 @@ public void onWillStartLoadingMap() {
     public void onDidFinishLoadingStyle() {
       if (mapboxMap != null) {
         if (initialLoad) {
+          initialLoad = false;
           mapboxMap.onPreMapReady();
           onMapReady();
           mapboxMap.onPostMapReady();
         } else {
           mapboxMap.onFinishLoadingStyle();
         }
       }
-      initialLoad = false;
```

Comment:
This flag update needs to stay, to mark, that the style has already been loaded before the rendering surface has been created (therefore, `mapboxMap` here would've been `null`), by for ex. calling `MapView#setStyleUrl`. Then the logic will follow the path of `MapCallback#initialised`. I think having the flag update in both places should be fine.